src/controller/CommissioningDelegate.h

```diff
@@ -555,7 +553,6 @@ class CommissioningParameters
         return *this;
     }
 
-#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
```

Comment:
This used to be conditional so that API consumers would easily discover that doing `SetUseJCM(true)` when CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC is not defined does not actually work.  Why is this being made unconditional?